### PR TITLE
Add selected props to radio widgets

### PR DESCRIPTION
**Pull request by Mottie**
## Description

When a radio button is selected, we sometimes show additional information (see screenshot). To get screenreaders to include the extra information, we associate the radio input with the alert ID using `aria-describedby`; but, the `aria-describedby` can not contain an ID that does not yet exist on the page, so we need to dynamically add the attribute when the user selects it.

The existing `ui:options.widgetProps` setting will add the props unconditionally, so a new `selectedProps` was added in this PR which conditionally adds props when a radio is selected.

Related:
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/19733
- https://github.com/department-of-veterans-affairs/veteran-facing-services-tools/pull/604

## Testing done

Updated unit tests

## Screenshots

<details><summary>Alert displayed after choosing "No"</summary>

Screen Shot 2021-04-02 at 1 23 24 PM</details>

<details><summary>axe error when <code>aria-describedby</code> is unconditionally added</summary>

Screen Shot 2021-04-02 at 10 36 58 AM</details>

## Acceptance criteria
- [x] Conditionally add radio attributes

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by joshkimux — APPROVED** [review]
Nice work! I'll test this out once it gets to staging 👍

File: src/platform/forms-system/test/js/widgets/RadioWidget.unit.spec.jsx [hunk 1]
```diff
@@ -136,4 +137,56 @@ describe('Schemaform <RadioWidget>', () => {
...
+      <RadioWidget value="1" onChange={onChange} options={options} />,
+    );
+    const inputs = tree.everySubTree('input');
+    expect(inputs[0].props['data-test']).to.equal('first');
+    expect(inputs[0].props['data-selected']).to.equal('first');
+
+    expect(inputs[1].props['data-test']).to.equal('second');
+    expect(inputs[1].props['data-selected']).to.be.undefined;
+  });
+  it('should add custom and selected props', () => {
```
Comment (bkjohnson) [inline src/platform/forms-system/test/js/widgets/RadioWidget.unit.spec.jsx:166]: Is the only difference between this and the previous test the `value` being 2? If so, can that be communicated a bit more, or the tests restructured?

**Review by bkjohnson — COMMENTED** [review]
Thank you for updating the documentation! I left a question there as well.

File: src/platform/forms-system/test/js/widgets/RadioWidget.unit.spec.jsx (hunk 1 above)
Comment (Mottie) [reply src/platform/forms-system/test/js/widgets/RadioWidget.unit.spec.jsx]: I've updated the test to re-render with new values. Included changes in the `YesNoWidget` test as well.

**Review by bkjohnson — APPROVED** [review]

**Pull request merged**